lib/mysql2/client.rb

```diff
@@ -87,6 +92,23 @@ def initialize(opts = {})
       connect user, pass, host, port, database, socket, flags
     end
 
+    def parse_ssl_mode( m )
+      return nil if m.nil?
+      if m.is_a?( String ) || m.is_a?( Symbol )
+        m.to_s.upcase!
```

Comment:
I've made your changes, that will remove the ability to pass in an integer, ala `ssl_mode: Mysql2::Client::SSL_MODE_DISABLED`.